Implement object verification process.

mithrandi opened the pull request:
Fixes #26.

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/fusionapp/entropy/28)
<!-- Reviewable:end -->

jonathanj commented:
This mostly seems okay, there are a few inline comments and some apparent missing test cases.
<!-- Reviewable comment -K3JMUvX665nc5xhKmYk:108101925 -->

---

Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

<sup>**[entropy/store.py, line 137 \[r1\]](https://reviewable.io:443/reviews/fusionapp/entropy/28#-K3JJ0tuzukgi37mE0vl)** ([raw file](https://github.com/fusionapp/entropy/blob/475f0abaaccd1bbdc0d7665d431c9da40390b74b/entropy/store.py#L137)):</sup>
This docstring doesn't document the conditions under which exceptions are raised.
<!-- Reviewable comment -K3JJ0tuzukgi37mE0vm:-936137729 -->

---

<sup>**[entropy/store.py, line 168 \[r1\]](https://reviewable.io:443/reviews/fusionapp/entropy/28#-K3JJ8WWQKgkM2PJqMpb)** ([raw file](https://github.com/fusionapp/entropy/blob/475f0abaaccd1bbdc0d7665d431c9da40390b74b/entropy/store.py#L168)):</sup>
It's a little weird that this exception is used for both possible cases of "bad stuff" even though they mean quite different things.
<!-- Reviewable comment -K3JJ8WWQKgkM2PJqMpc:-1627678403 -->

---

<sup>**[entropy/store.py, line 168 \[r1\]](https://reviewable.io:443/reviews/fusionapp/entropy/28#-K3JLgz79MomwAF1ZmbZ)** ([raw file](https://github.com/fusionapp/entropy/blob/475f0abaaccd1bbdc0d7665d431c9da40390b74b/entropy/store.py#L168)):</sup>
There doesn't seem to be a testcase for this mode of failure.
<!-- Reviewable comment -K3JLgz79MomwAF1Zmb_:1194748187 -->

---

<sup>**[entropy/test/test_store.py, line 821 \[r1\]](https://reviewable.io:443/reviews/fusionapp/entropy/28#-K3JLPEF8q1kEO7HSMyS)** ([raw file](https://github.com/fusionapp/entropy/blob/475f0abaaccd1bbdc0d7665d431c9da40390b74b/entropy/test/test_store.py#L821)):</sup>
It might be worth asserting something about the saved object ID.
<!-- Reviewable comment -K3JLPEF8q1kEO7HSMyT:-113452051 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/fusionapp/entropy/28)
<!-- Sent from Reviewable.io -->

mithrandi commented:
Review status: 2 of 5 files reviewed at latest revision, 4 unresolved discussions.

---

_[entropy/store.py, line 137 \[r1\]](https://reviewable.io:443/reviews/fusionapp/entropy/28#-K3JJ0tuzukgi37mE0vl:-K3SSDXpN1n801VLtiFr:286763122) ([raw file](https://github.com/fusionapp/entropy/blob/475f0abaaccd1bbdc0d7665d431c9da40390b74b/entropy/store.py#L137)):_
I've added `@raise` documentation.

---

_[entropy/store.py, line 168 \[r1\]](https://reviewable.io:443/reviews/fusionapp/entropy/28#-K3JJ8WWQKgkM2PJqMpb:-K3SSP375-Z7hWxEg1W_:-475051072) ([raw file](https://github.com/fusionapp/entropy/blob/475f0abaaccd1bbdc0d7665d431c9da40390b74b/entropy/store.py#L168)):_
There are now two different subclasses for the two weird cases.

---

_[entropy/store.py, line 168 \[r1\]](https://reviewable.io:443/reviews/fusionapp/entropy/28#-K3JLgz79MomwAF1ZmbZ:-K3STx5VF8IRmWS9Vc52:-1652431528) ([raw file](https://github.com/fusionapp/entropy/blob/475f0abaaccd1bbdc0d7665d431c9da40390b74b/entropy/store.py#L168)):_
I've added a test for this exception path now.

---

_[entropy/test/test_store.py, line 821 \[r1\]](https://reviewable.io:443/reviews/fusionapp/entropy/28#-K3JLPEF8q1kEO7HSMyS:-K3SUQI3a8RUl81mGh5Q:-499146009) ([raw file](https://github.com/fusionapp/entropy/blob/475f0abaaccd1bbdc0d7665d431c9da40390b74b/entropy/test/test_store.py#L821)):_
Done.

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/fusionapp/entropy/28)_
<!-- Sent from Reviewable.io -->

jonathanj commented:
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/fusionapp/entropy/28)_
<!-- Sent from Reviewable.io -->
